### DISCUSSION TITLE
Scope/Structure of the curriculum (What makes a metaverse dev?)

### DISCUSSION THREAD

**Discussion by msub2** [AngellXR/metaversedev#17] [Ideas]
I think one of the most important discussions to have here is deciding exactly *what* it is that would be taught. I'll start by referencing the initial "prototype" homepage for the previous metaversity effort: https://metaversity.foundation

So the proposed offering of the entire program was
- Code fundamentals
- Creating 3D content (modeling, rigging, texturing, animating, etc with Blender)
- Creating XR content (the proposed lesson plan seems to favor WebXR with three.js specifically)
- Blockchain (General concepts and Ethereum-specific smart contract programming w/ Solidity)
- AI (general concepts, deep learning, training models, "digital beings")
- Ethics (an interesting inclusion among the rest for sure, I don't know of any course out there that specifically covers ethics)

This is a *lot* of content. Potentially hundreds of hours worth depending on the depth the curriculum goes into. So what should people expect to be getting out of it, and how much is content that metaverse.dev would be expected to cover? Part of me is wary of having to develop, say, an introduction to coding or an introduction to Blender when these resources are plentiful on the internet already. If the curriculum for these tracks is going to be original, each of these subjects would need multiple subject experts to determine what to cover and how to craft an appropriate lesson plan that teaches potential students what they need to know.

I think what would make the most sense is (assuming metaverse.dev follows an open curriculum plan similar to The Odin Project or freeCodeCamp) offering these subjects as tracks, where one could follow the path that makes the most sense for them and receive some kind of certification at the end showing that they've completed it. Where applicable, tracks could branch out into more specific specializations (i.e. "demonstrated proficiency in A-Frame/babylon.js/three.js/Wonderland Engine/etc"), with an overall certification for those that complete some threshold of certifications in each of the offered areas (this would be somewhat similar to how freeCodeCamp gave out its Full Stack Developer certification once you complete the Front End, Back End, and Data Viz courses)

This is mostly spitballing from me though, curious to hear what other people think

**Comment by mrmetaverse** [comment]
Just realized this is totally related: https://github.com/AngellXR/metaversedev/discussions/11

**Comment by mrmetaverse** [comment]
I love the idea of a self-paced structure like The Odin Project. I also like the idea of a dev.to like structure where it's an open library for sharing and learning from others. A clean tagging structure for generating pages based on content tag, is pretty straightforward, and intuitive as a default interface. Admittedly the Odin project is more of a single structured curriculum whereas dev.to or even udemy are more knowledge bazaars of various topics. I'd like to allow "almost anyone" to add content. 

Viewers could follow people, platforms, or topics based on tag. I have a lot of experience with teams working in this way, but also understand the complexities of creating our own content management system. I'm open to lightweight and fast, so I'll call out that what I've described is lightweight and fast from a content creation and curation standpoint but may be a much heavier lift from a feasibility or scale side if we roll our own. If we buy off the shelf it can get pricey. I'd like to find a good fit for us. 

I even [like these quick shares from people like vrhermit:](https://twitter.com/vrhermit/status/1524816769583562768?s=20&t=XvjWh_joIdVY8X3TOueUPQ)

**Reply by mrmetaverse** [reply]
Found Lichen*, a new [lightweight CMS called Lichen](https://lichen.sensorstation.co/) that would be more than enough for a prototype. I don't consider the one your referenced as a prototype for our offering but it was a prototype made by a person who worked on the metaversity efforts. I am all for rapid experimentation and by no means attached to that example that was shared before. 

Here is an example site using the CMS as well:https://ediblegoosecreek.com/

**Reply by mrmetaverse** [reply]
[Papers with Code](https://paperswithcode.com/latest) is an interesting and effective way of sharing knowledge. The report, and the code, are provided together. Maybe something like this could work for some people who have created something new. micro lessons walking through the readme of their creation, and how to start the project and make your first edit. Then that author can add content at their own leisure, either expanding and gaining in complexity, or covering a wide array of basics, etc.

**Comment by mrmetaverse** [comment]
Michael used to work for Pluralsight and shared this as a comp https://www.pluralsight.com/ they used to work there

**Comment by mrmetaverse** [comment]
Some kind of rating system for the things that people submit. People can submit guides, and if people find certain guides to be helpful, they can upvote that guide. This can bubble up higher quality offerings. 

Q: What does hackernews do? Is it number of upvotes in a certain time interval?

**Comment by mrmetaverse** [comment]
Lets create a first or first two curriculums and then open it up to the public to submit how to guides. 
- becoming a threejs 
- becoming a 3d modeler 

- how to develop with webaverse 
- how to model clothing for avatars 

Q: How do we onboard new content? 
Submit a pull request.

**Comment by mrmetaverse** [comment]
What are things we could work on async over the next month? 
- this would have been nice to have more people here today. Lets plan on a meeting with more people in attendance (maybe a workshop to ideate MVP) 
  - like would a simple github pages page work for now? or are we still so new. 
- can we increase meeting cadence? 
- what does a first curriculum look like? Msub2 has enough webxr experience to put together an alpha curriculum.

**Reply by mrmetaverse** [reply]
_here is how to get a basic game live with webxr_

**Reply by mrmetaverse** [reply]
consider primer tutorials, i.e. a Github primer, webpack primer, etc.

**Comment by mrmetaverse** [comment]
RE: assuming leadership roles
- Daniel msub2 volunteers to assume some of these roles.

**Comment by mrmetaverse** [comment]
WOW this is effective formatting
https://twitter.com/JanvandenHemel/status/1545029871524462594?s=20&t=8iVCS2HWeATI1ljFrcRmbg

**Comment by mrmetaverse** [comment]
https://twitter.com/JanvandenHemel/status/1548716900540104706

**Comment by mrmetaverse** [comment]
~~Maybe we get~~ I made a spreadsheet going of interesting reference content. https://github.com/AngellXR/metaversedev/issues/24

**Comment by mrmetaverse** [comment]
https://twitter.com/denicmarko/status/1550087191254138881?s=20&t=IcTKssrbkyGq2yhsKwPJdA

![Screenshot_20220723-070213.png](https://user-images.githubusercontent.com/63426722/180602340-c8f1c094-7f67-4ace-9769-3160805f9c25.png)

**Comment by lalalune** [comment]
I think that to start off, focusing on solution-oriented tutorials would be a good way to show numbers and drive hits.

For example -- "how to make an avatar for VRChat".

I think you can build more complex curriculum and tracks but it's good to start with microlessons where users are coming, getting a skill or sharing a concise way to acquire the skill with their following.

I like what @mrmetaverse pointed to -- bitesize twitter/insta videos with the messaging in text, then a link to the full tutorial or video would be pretty legit.

**Comment by mrmetaverse** [comment]
WOW 
https://offscreencanvas.com/issues/008
